extensions/wikia/Search/classes/Result.php

```diff
@@ -205,53 +242,76 @@ public function getVars() {
 
 	/**
 	 * Removes junk values from snippeting
+	 *
 	 * @see   WikiaSearchResultTest::testFixSnippeting
-	 * @param string  $text
+	 *
+	 * @param string $text
 	 * @param boolean $addEllipses
+	 *
+	 * @return string
 	 */
-	private static function fixSnippeting($text, $addEllipses=false) {
-		$text = preg_replace('/^(span class="searchmatch">)/', '<$1',
-			preg_replace("/^[[:punct:]]+ ?/", '',
-				preg_replace("/(<\\/span>)('s)/i", '$2$1',
-					preg_replace('/ +$/', '',
-						preg_replace('/ ?\.{2,3}$/', '',
-							preg_replace('/ ?&hellip;$/', '',
-								str_replace('�', '', $text)))))));
-		$text = strlen($text) > 0 && $addEllipses
-			? preg_replace('/(<\/span)$/', '$1>', preg_replace('/[[:punct:]]+$/', '', $text)).'&hellip;'
-			: $text;
+	private static function fixSnippeting( $text, $addEllipses = false ) {
+		$text = preg_replace(
+			'/^(span class="searchmatch">)/',
+			'<$1',
+			preg_replace(
+				"/^[[:punct:]]+ ?/",
+				'',
+				preg_replace(
+					"/(<\\/span>)('s)/i",
+					'$2$1',
+					preg_replace(
+						'/ +$/',
+						'',
+						preg_replace(
+							'/ ?\.{2,3}$/',
+							'',
+							preg_replace(
+								'/ ?&hellip;$/',
+								'',
+								str_replace( '�', '', $text )
+							)
+						)
+					)
+				)
+			)
+		);
+		$text = strlen( $text ) > 0 && $addEllipses ?
+			preg_replace( '/(<\/span)$/', '$1>', preg_replace( '/[[:punct:]]+$/', '', $text ) ) . '&hellip;' : $text;
 		$text = strip_tags( $text, '<span>' );
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Immediately return this expression instead of assigning it to the temporary variable "$text". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1488)